packages/react-components/react-button/src/components/Button/useButton.ts

```diff
@@ -35,19 +35,13 @@ export const useButton_unstable = (
     size, // State calculated from a set of props
     iconOnly: Boolean(iconShorthand?.children && !props.children), // Slots definition
     components: { root: 'button', icon: 'span' },
-    root: slot.always(
-      getIntrinsicElementProps(
-        as,
-        useARIAButtonShorthand<ARIAButtonSlotProps<'a'>>(props, {
-          required: true,
-          defaultProps: {
-            ref: ref as React.Ref<HTMLButtonElement & HTMLAnchorElement>,
-            type: 'button',
-          },
-        }),
-      ),
-      { elementType: 'button' },
-    ),
+    root: slot.always<ARIAButtonSlotProps<'a'>>(getIntrinsicElementProps(as, useARIAButtonProps(props.as, props)), {
+      elementType: 'button',
+      defaultProps: {
+        ref: ref as React.Ref<HTMLButtonElement & HTMLAnchorElement>,
+        type: 'button',
+      },
+    }),
```

Comment:
Based on how it's done elsewhere in this PR, shouldn't you resolve the slot with it's default props and then send that to `useARIAButtonProps` as `useARIAButtonProps(rootSlot.as, root)`? Or is it different here because `root` is the primary slot?

Reply:
In this particular case, the order doesn't matter, since there's no custom property being declared in the `defaultProps` (like `disabledFocusable` for example), but for the sake of a pattern I'll update it to follow up how it's done in other components!

Reply:
Since it's `root` the usage of `useARIAButtonProps` should be first as otherwise the custom properties provided by the user will be filtered out by `getIntrinsicElementProps`